site/en/articles/new-headless/index.md

```diff
@@ -0,0 +1,204 @@
+
+```html
+<output>0</output>
+<script>
+  const element = document.querySelector('output');
+  let counter = 0;
+  setInterval(() => {
+    counter++;
+    console.log(counter);
+    element.textContent = counter;
+  }, 1_000);
+</script>
+```
+
+After one second, the page contains “1”; after two seconds, “2”, and so on. Here’s how you’d capture the page’s state after 5 seconds and save it as a PDF:
+
+```sh
+chrome --headless=new --print-to-pdf --timeout=5000 https://mathiasbynens.be/demo/time
+```
+
+The `--timeout=5000` flag tells Chrome to wait for 5 seconds before printing the PDF. Thus, this process takes at least 5 seconds to run.
+
+#### `--virtual-time-budget`
+
+The `--virtual-time-budget` enables time travel! Well, kinda. Virtual Time acts as a “fast-forward” for any time-dependent code (e.g. `setTimeout`/`setInterval`). It forces the browser to execute any of the page’s code as fast as possible while making the page believe that the time actually goes by.
+
+`--virtual-time-budget` is most commonly used as a replacement for `--timeout`, like this:
+
+```sh
+chrome --headless=new --print-to-pdf --virtual-time-budget=5000 https://mathiasbynens.be/demo/time
+```
+
+This produces the same result as the previous example with `--timeout`. The difference is that with `--virtual-time-budget` the process takes very little real time — just about the same time as if there was no timeout specified at all. The difference becomes more obvious with larger values:
+
+```sh
+chrome --headless=new --print-to-pdf --virtual-time-budget=42000 https://mathiasbynens.be/demo/time
+```
+
+With `--timeout=42000` it would take at least 42 seconds before the PDF gets printed. With `--virtual-time-budget=42000` it barely takes longer than the previous example.
+
+## Debugging
+
+Since Chrome is effectively invisible in Headless mode, it might sound tricky to figure out what’s going on in case of issues. Luckily, it’s possible to debug Headless Chrome in a way that’s very similar to headful Chrome. The trick is to launch Chrome in Headless mode with the `--remote-debugging-port` command-line flag.
+
+```sh
+chrome --headless=new --remote-debugging-port=0 https://developer.chrome.com/
+```
+
+This prints a unique WebSocket URL to stdout, for example:
+
+```txt
+DevTools listening on ws://127.0.0.1:60926/devtools/browser/b4bd6eaa-b7c8-4319-8212-225097472fd9
+```
+
+In a regular headful Chrome instance, we can then use [Chrome DevTools remote debugging](/docs/devtools/remote-debugging/) to connect to the Headless target and inspect it. To do so, go to `chrome://inspect`, click the “Configure…” button, and enter the IP address and port number from the WebSocket URL. In the above example, I entered `127.0.0.1:60926`. Click “Done”, and you should see a Remote Target appear with all its tabs and other targets listed below. Click “inspect”, _et voilà_ — you now have access to Chrome DevTools inspecting the remote Headless target, **including a live view of the page**!
```

Comment:
```suggestion
In a regular headful Chrome instance, we can then use [Chrome DevTools remote debugging](/docs/devtools/remote-debugging/) to connect to the Headless target and inspect it. To do so, go to `chrome://inspect`, click the **Configure…** button, and enter the IP address and port number from the WebSocket URL. In the above example, I entered `127.0.0.1:60926`. Click **Done** and you should see a Remote Target appear with all its tabs and other targets listed below. Click **inspect** and you now have access to Chrome DevTools inspecting the remote Headless target, **including a live view of the page**!
```
Removed "et voila" because a11ty. Unfortunately, we can't expect non-native Eng speakers to also understand even a bit of French.
Voila is also not to use according to the style guide: https://developers.google.com/style/word-list#:~:text=See%20also%20interconnectAttachment.-,voila

Reply:
Merci beaucoup!